config/routes.rb

```diff
@@ -1,4 +1,14 @@
+class AuthConstraint
+  def matches?(request)
+    request.session[:user]["email"].present?
```

Comment:
one cool thing I learned recently is that rails automatically encrypts session tokens for you

> To prevent session hash tampering, a digest is calculated from the session with a server-side secret (secrets.secret_token) and inserted into the end of the cookie.

http://guides.rubyonrails.org/security.html#session-storage

so I think storing email is fine -- when we look it up it also automatically de-crypts so looking up user by email should be fine.

btw- learned this while having a very similar convo about `current_user` and security in C2 
https://github.com/18F/C2/blob/master/app/controllers/application_controller.rb#L118